x/session/keeper/session_hydrator.go

```diff
@@ -241,6 +233,20 @@ func (k Keeper) hydrateSessionSuppliers(ctx context.Context, sh *sessionHydrator
 	return nil
 }
 
+// Generate deterministic random weight for supplier:
+// 1. Combine session ID and supplier's operator address to create unique seed
+// 2. Hash the seed using SHA3-256
+// 3. Take first 8 bytes of hash as random weight
+func generateSupplierRandomWeight(supplier *sharedtypes.Supplier, sh *sessionHydrator) int {
```

Comment:
One small request. 

Can you do one of the following:

1. Either only pass in `sh.sessionIDBz` instead of the entire hydrator
OR
2. Move this helper to where we define `sessionHydrator` and have it be a receiver so we can do `sh.GenerateSupplierRandomWeight(supplier)`